modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/RDBMSCoordinationStrategy.java

```diff
@@ -212,6 +212,23 @@ public void start(CoordinationConfigurableClusterAgent configurableClusterAgent,
 
         coordinatorElectionTask = new CoordinatorElectionTask();
         threadExecutor.execute(coordinatorElectionTask);
+
+        // Wait until node state become Candidate/Coordinator because thrift server needs to start after that.
+        int waitTime = 0;
+        int maxWaitTime = heartbeatMaxAge * 5;
+        while (currentNodeState == NodeState.ELECTION) {
+            try {
+                TimeUnit.MILLISECONDS.sleep(500);
+                waitTime = waitTime + 500;
+                if (waitTime == maxWaitTime) {
+                    throw new RuntimeException("Node status didn't change to Candidate/Coordinator within "
+                            + waitTime + " milliseconds.");
+                }
+            } catch (InterruptedException e) {
+                logger.error("An error occurred while waiting to get current node state.", e);
+                Thread.currentThread().interrupt();
+            }
+        }
```

Comment:
As per above comment, code changed to throw RuntimeException if InterruptedException occurred.  I feel that it is not necessary to move it to private method since there is no code duplication.